fossunited/api/profile.py

```diff
@@ -56,63 +56,89 @@ def toggle_profile_privacy(value):
+            "medium": fields_dict.get("medium"),
+            "mastodon": fields_dict.get("mastodon"),
+        }
+
         frappe.db.set_value(
-            USER_PROFILE,
-            user_doc.name,
-            {
-                "full_name": fields_dict.get("full_name"),
-                "username": fields_dict.get("username"),
-                "route": fields_dict.get("username"),
-                "bio": fields_dict.get("bio"),
-                "current_city": fields_dict.get("current_city"),
-                "about": fields_dict.get("about"),
-                "website": fields_dict.get("website"),
-                "x": fields_dict.get("x"),
-                "linkedin": fields_dict.get("linkedin"),
-                "github": fields_dict.get("github"),
-                "gitlab": fields_dict.get("gitlab"),
-                "instagram": fields_dict.get("instagram"),
-                "youtube": fields_dict.get("youtube"),
-                "devto": fields_dict.get("devto"),
-                "medium": fields_dict.get("medium"),
-                "mastodon": fields_dict.get("mastodon"),
-            },
+            USER_PROFILE, user_doc.name, updated_fields
         )
+
+        if (
+            fields_dict.get("full_name") != user_doc.full_name
+            or fields_dict.get("username") != user_doc.username
+        ):
+            user_updates = {}
+            if fields_dict.get("full_name") != user_doc.full_name:
+                user_updates["full_name"] = fields_dict.get(
+                    "full_name"
+                )
+
+            if fields_dict.get("username") != user_doc.username:
+                user_updates["username"] = fields_dict.get("username")
+
+            if user_updates:
+                frappe.db.set_value(
+                    "User", user_doc.user, user_updates
+                )
+
+        return True
+
     except Exception as e:
-        frappe.throw(str(e))
+        frappe.log_error(
+            ("Error updating profile: {0}").format(str(e))
+        )
+        frappe.throw(
+            (
+                "An error occurred while updating the profile. Please try again."
+            )
+        )
 
 
 @frappe.whitelist()
-def is_unique_username(username: str, id: str) -> bool:
+def is_valid_username(username: str, id: str) -> bool:
     """
-    Check if the username is unique
+    Check if the username is unique and not in restricted list
 
     Args:
         username: Username to check
         id: ID of the user profile
 
     Returns:
-        bool: True if username is unique
+        bool: True if username is unique and not a restricted username
     """
     if (
-        frappe.db.exists("FOSS Event Chapter", {"route": username})
+        frappe.db.exists("FOSS Chapter", {"route": username})
         or frappe.db.exists(
             USER_PROFILE,
             {"route": username, "name": ["!=", id]},
         )
         or frappe.db.exists(
-            "User",
+            RESTRICTED_USERNAME,
```

Comment:
Is the Restricted Username doctype there already? I can't see it's creation in this PR

Reply:
Ahh. 

I forgot that I had closed the other PR in. The doctype was in my local but not in the files. 

Adding it in.